compiler/qsc_linter/src/lints.rs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+pub(super) mod ast;
+pub(super) mod hir;
+
+macro_rules! push_lint {
+    ($lint:expr, $span:expr, $buffer:expr) => {
+        $buffer.push(Lint {
```

Comment:
Just personal preference but I think this macro obfuscates a little too much. I'd find it easier to follow `buffer.push(lint!(self, expr.span))` at the call site, and it's not any longer than writing `push_lint!(self, expr.span, buffer)`.

Reply:
I should have clarified that the reason I'm averse to macros is that the editor experience (code navigation, hover info, parameter hints etc) for macros tends to be pretty terrible, so in general I think they should only be used if they really help save _a lot_ of code.

Reply:
Yes, I agree with your preference haha.  Just changed it as you suggested.